table/server/master/src/main/java/alluxio/master/table/Database.java

```diff
@@ -410,6 +419,38 @@ private boolean applyAddTable(@Nullable JournalContext context, Journal.JournalE
     return true;
   }
 
+  private boolean applyAddTablePartitions(@Nullable JournalContext context,
+      Journal.JournalEntry entry) {
+    alluxio.proto.journal.Table.AddTablePartitionsEntry addTablePartitions
+        = entry.getAddTablePartitions();
+    if (!addTablePartitions.getDbName().equals(mName)) {
+      return false;
+    }
+
+    mTables.compute(addTablePartitions.getTableName(), (key, existingTable) -> {
+      if (existingTable != null) {
+        if (addTablePartitions.getVersion() == existingTable.getVersion()) {
+          // this table is being removed, and has the expected version
+          LOG.info("Adding partitions to table {}.{}", mName, addTablePartitions.getTableName());
```

Comment:
can we also add the # partitions we are adding in the message?

Reply:
Added, thanks